tests/new_upgrades/test_activation_key.py

```diff
@@ -0,0 +1,129 @@
+"""Test Activation Key related Upgrade Scenario's
+
+:Requirement: UpgradedSatellite
+
+:CaseAutomation: Automated
+
+:CaseComponent: ActivationKeys
+
+:Team: Phoenix-subscriptions
+
+:CaseImportance: High
+
+"""
+
+from box import Box
+from fauxfactory import gen_alpha
+import pytest
+
+from robottelo.config import settings
+from requests.exceptions import HTTPError
+from robottelo.utils.shared_resource import SharedResource
+
+
+@pytest.fixture
+def ak_upgrade_setup(content_upgrade_shared_satellite, upgrade_action):
+    """Pre-upgrade scenario that creates an activation key and custom repo.
+    
+    :id: preupgrade-a7443b54-eb2e-497b-8a50-92abeae01496
+    
+    :steps:
+        1. Create organization.
+        2. Create product.
+        3. Create and sync custom repo.
+        4. Create activation key.
+    """
+    target_sat = content_upgrade_shared_satellite
+    with SharedResource(target_sat.hostname, upgrade_action, target_sat=target_sat) as sat_upgrade:
+        test_data = Box(
+            {
+                'target_sat': target_sat,
+                'test_name': None,
+                'org': None,
+                'product': None,
+                'repo': None,
+                'cv': None,
+                'ak': None,
+            }
+        )
+        test_name = f'ak_upgrade_{gen_alpha()}'
+        test_data.test_name = test_name
+        org = target_sat.api.Organization(name=f"{test_name}_org").create()
+        test_data.org = org
+        product = target_sat.api.Product(organization=org, name=f'{test_name}_prod').create()
+        test_data.product = product
+        custom_repo = target_sat.api.Repository(
+            product = product,
+            name = f'{test_name}_repo',
+            url=settings.repos.yum_1.url,
+            content_type = 'yum',
+        ).create()
+        test_data.repo = custom_repo
+        custom_repo.sync()
+        cv = target_sat.publish_content_view(org, [custom_repo], f'{test_name}_cv')
+        assert len(cv.read_json()['versions']) == 1
+        test_data.cv = cv
+        ak = target_sat.api.ActivationKey(
+            content_view=cv, organization=org, name=f'{test_name}_ak'
+        ).create()
```

Comment:
This creation should fail since 6.17 (both, CV and LCE, is required). Could we make this creation a bit future-proof by providing it with the Library LCE?
```suggestion
        ak = target_sat.api.ActivationKey(
            content_view=cv, environment=org.library.id, organization=org, name=f'{test_name}_ak'
        ).create()
```

Reply:
@vsedmik Sure thing, thanks for the feedback. My first thought was that we might want to create a new LCE, but since the post-upgrade section deletes the org at the end, I don't think that's actually necessary.

Reply:
@synkd That was my first thought too, but didn't want to bother with that and CV promotion. Library will do job too with less pain.